src/app/shared/ui-kit/ui-question-card/ui-question-card.component.ts

```diff
@@ -59,6 +59,10 @@ export class UiQuestionCardComponent implements OnInit {
     this.numberOfQuestions = +this.route.snapshot.paramMap.get('questions')!;
 
     this.loadQuestion();
+
+    this.radioButtonControl.valueChanges.subscribe(() => {
+      this.isMessageVisible = false;
+    });
```

Comment:
this part of code is not obvious, difficult to understand why it should be present. I'd rather add a separate method resetStepMessage(): void or something like that, and invoke it in OnInit hook.
And I believe we need to unsubscribe also